src/server/editorServices.ts

```diff
@@ -502,7 +522,11 @@ namespace ts.server {
             return scriptInfo && scriptInfo.getDefaultProject();
         }
 
-        private ensureInferredProjectsUpToDate() {
+        /**
+         * Ensures the project structures are upto date
+         * @param refreshInferredProjects when true updates the inferred projects even if there is no pending work
+         */
+        private ensureInferredProjectsUpToDate(refreshInferredProjects?: boolean) {
             if (this.changedFiles) {
                 let projectsToUpdate: Project[];

```

Comment:
I have renamed this function to ensureProjectStructuresUptoDate just so that it is clear what this is achieving. Also added the note as:
```TypeScript
/**
         * Ensures the project structures are upto date
         * This means,
         * - if there are changedFiles (the files were updated but their containing project graph was not upto date),
         *   their project graph is updated
         * - If there are pendingProjectUpdates (scheduled to be updated with delay so they can batch update the graph if there are several changes in short time span)
         *   their project graph is updated
         * - If there were project graph updates and/or there was pending inferred project update and/or called forced the inferred project structure refresh
         *   Inferred projects are created/updated/deleted based on open files states
         * @param forceInferredProjectsRefresh when true updates the inferred projects even if there is no pending work to update the files/project structures
         */
        private ensureProjectStructuresUptoDate(forceInferredProjectsRefresh?: boolean) {

```